### DISCUSSION TITLE
Filter numeric values as strings

### DISCUSSION THREAD

**Discussion by Asky-GH** [yahoo/elide#1900] [Q&A] [answered]
I have some data with numeric values like, for instance, age etc. And I want to be able to filter such values using operators as LIKE, IN etc. Is it possible?

**Comment by aklish** [comment, marked as answer]
Looking at the code, Elide allows the IN operator to be used with numeric values (not INI - or case insensitive IN).  For INFIX, PREFIX, and POSTFIX, Elide will try to coerce the type to a String.  I don't see any test cases for this though.

**Reply by aklish** [reply]
Added an issue for this:

https://github.com/yahoo/elide/issues/1907

**Reply by aklish** [reply]
Added a few tests here:

https://github.com/yahoo/elide/pull/1908/files

The tested operators work on numeric values.

**Reply by Asky-GH** [reply]
Interesting. I'm using Elide 4 and when I try something like this "/entity?filter[entity]=userId==\*2\*" I get the following error:
Caused by: org.postgresql.util.PSQLException: ERROR: function lower(bigint) does not exist

**Reply by aklish** [reply]
There was a change in Elide 5 with respect to FIQL and case sensitivity:

```
FIQL operators are now case sensitive by default. New case insensitive operators have been introduced allowing greater flexibility. It is possible to revert to elide 4 semantics through configuration.
```

This was the PR that enabled this change: https://github.com/yahoo/elide/pull/1519

If you don't need case insensitive IN and NOT IN, you can do something similar in Elide 4 by configuring the RSQLFilterDialect with the following case sensitivity strategy:
```CaseSensitivityStrategy.UseColumnCollation()```

Here is a snippet in the Spring bean where the RSQL Dialect is configured:
https://elide.io/pages/guide/v5/10-jsonapi.html#fiql-default-behaviour
